src/vs/base/test/node/encoding/encoding.test.ts

```diff
@@ -231,30 +232,32 @@ suite('Encoding', () => {
 		});
 	}
 
-	async function readAllAsString(stream: NodeJS.ReadableStream) {
-		return new Promise<string>((resolve, reject) => {
-			let all = '';
-			stream.on('data', chunk => {
-				all += chunk;
-				assert.equal(typeof chunk, 'string');
-			});
-			stream.on('end', () => {
-				resolve(all);
+	function newTestReadableStream(buffers: Buffer[]): VSBufferReadableStream {
+		const stream = newWriteableBufferStream();
+		buffers
+			.map(VSBuffer.wrap)
+			.forEach(buffer => {
+				setTimeout(() => {
+					stream.write(buffer);
+				});
 			});
-			stream.on('error', reject);
+		setTimeout(() => {
+			stream.end();
 		});
+		return stream;
+	}
+
+	async function readAllAsString(stream: streams.ReadableStream<string>) {
+		return streams.consumeStream(stream, strings => strings.join(''));
```

Comment:
A bit offtopic:

I was wondering why do you have to pass `reducer` to `consumeStream`? It is already being passed on stream creation. Why not reuse it?

Reply:
@gyzerok we still need the reducer because `consumeStream` is generic, right? Or would you suggest to have the `reducer` be accessible from the stream to reuse it?

Reply:
Yeah, I was thinking more conceptually, not from the types perspective. At least from my point of view it makes sense to change it so, that the reducer passed on the creation is reused. I assume it can either be done by either making `consumeStream` a method of `ReadableStream` or by making reducer accessible.

Anyway, consider it more as a suggestion from the "new to the project" person perspective :)

Reply:
I am trying very hard to have as little methods on the stream as possible to allow for easy inter-op with e.g. node.js or any other stream interface. So I do not want to push too many things onto it.